rust/cjdns_sys/src/interface/wire/message.rs

```diff
@@ -0,0 +1,145 @@
+//! Message type.
+
+use thiserror::Error;
+
+/// A network message.
+///
+/// Can be converted to/from to C `struct Message` from `wire/Message.h`.
+#[derive(Clone)]
+pub struct Message {
+    /// Actual data starts at offset `padding`.
+    data: Box<[u8]>,
```

Comment:
I would rather have Message wrap `ffi::Message`, we can expect that every message will be created in C (for now) and it allows `from_c_message()` to be zero-copy

Reply:
@cjdelisle I agree on you

Reply:
This doesn't make a good API because after wrapping the original message pointer it must be kept valid while the wrapper is in use. I am personally strongly against such APIs, but if you prefer it this way - okay, it's your party. See the update.